### PR TITLE
Finalize fixes

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Comment by tysondenio** [comment]
Ok, just to make sure I'm understanding here...

The `event_handlers` that get passed in to `SubscriptionConsumer` are internal to the `SubscriptionConsumer`.  `SubscriptionConsumer.emit` is dispatching the messages to those `event_handlers` and does not call the user's supplied handler.  Dispatching the message to user-supplied code is still only initiated by the call to `handler` correct?

File: kafkaesk/schema.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ async def finalize(self) -> None:
         if self._producer is not None:
             await self._producer.stop()
```
Comment (tysondenio) [inline kafkaesk/schema.py:73]: Do we need to flush this producer?

Comment (vangheem) [reply kafkaesk/schema.py:74]: yup, good idea.

**Comment by vangheem** [comment]
Yes, the event handlers are right now mostly for `consume_for` which is used in tests.

We could expose it if we thought it was useful though eventually.

We might need to refactor all of this. It was the first pass at it but maybe we should redesign it.

**Comment by tysondenio** [comment]
Ok, I just wanted to make sure I knew what was going on.  I think the design is ok for now and I don't think we need to expose those yet.  If we end up coming up with more events to emit (more than just 'on_message') then that may be the time to take another look.

**Pull request merged**